src/router.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3.7
+
+import socket
+import binascii
+import struct
+from pascy.l2 import *
+from pascy.l3 import *
+from pascy.l4 import *
+from pascy.layer import *
+from pascy.fields import *
+from dataclasses import dataclass
+from collections import namedtuple
+
+
+@dataclass
+class RoutingTable:
+    """
+    Holds the routing table the a router is using. Enables 
+    to resolve the next interface to forward a packet. """
+
+    ENDIANITY = 'big'
+
+    routing_table: list
+    
+    def route(self, dest_ip):
+        '''
+        Determine the interface through which the router should 
+        forward packets to get to the destination ip address.
+
+        :param dest_ip: the ip address of the destination
+        :returns: the resolved interface ip
+        :rtype: string
+        '''
+        self.routing_table.sort(reverse=True, key= lambda entry : entry[1])
+        for entry in self.routing_table:
+            dest_ip_int = int.from_bytes(socket.inet_aton(dest_ip), RoutingTable.ENDIANITY)
+            mask_int = int.from_bytes(entry.mask, RoutingTable.ENDIANITY)
+            entry_subnet_int = int.from_bytes(socket.inet_aton(entry.subnet), RoutingTable.ENDIANITY)
+            if dest_ip_int & mask_int == entry_subnet_int:
+                return entry.interface_ip
+        return None
+    
+
+class Router:
+    """
+    Performs the main logic of a router. Forwards IP 
+    packets and responds to ARP requests """
+
+    DEFAULT = '0.0.0.0'
+    MASK_24_SUBNET = socket.inet_aton('255.255.255.0')
+    DEFAULT_MASK = socket.inet_aton('0.0.0.0')
+    NET1_NAME = 'net1'
+    NET2_NAME = 'net2'
+    NET1_IP = '1.1.1.1'
+    NET2_IP = '2.2.2.1'
+    CLIENT2_IP = '2.2.2.2'
+    CLIENT1_IP = '1.1.1.2'
+    
+    ETH_TYPE_ARP = b'\x08\x06'
+    ETH_TYPE_IPV4 = b'\x08\x00'
+    BUF_SIZE = 2048
+    GGP = 0x0003
+
+    MAC_ROUTER_NET1 = '02:42:36:E0:40:43'
+    MAC_ROUTER_NET2 = '02:42:00:33:82:46'
+    CLIENT1_MAC = '02:42:01:01:01:02'
+    CLIENT2_MAC = '02:42:02:02:02:02'
+
+    INTERFACE_MAC2NAME = {MAC_ROUTER_NET1: NET1_NAME,
+                        MAC_ROUTER_NET2: NET2_NAME}
+
+    INTERFACE_NAME2IP = {NET1_NAME: NET1_IP,
+                        NET2_NAME: NET2_IP}
+
+    def __init__(self):
+        Interface = namedtuple('Interface', ['ip', 'mac', 'name'])
```

Comment:
It’s better to create the type outside the function.

Reply:
bc you use it outside the init
it creates new one every init